src/framework/execution/__tests__/dependencyGraph.test.ts

```diff
@@ -0,0 +1,571 @@
+  test('should throw an error if a circular dependency is created', () => {
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        dependsOn: ['b'],
+        executionHandler: jest.fn(),
+      },
+      {
+        id: 'b',
+        name: 'b',
+        types: [],
+        dependsOn: ['a'],
+        executionHandler: jest.fn(),
+      },
+    ];
+
+    expect(() => buildStepDependencyGraph(steps)).toThrow(
+      /Dependency Cycle Found/,
+    );
+  });
+
+  test('should throw an error a dependency that does not exist is specified', () => {
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        dependsOn: ['b'],
+        executionHandler: jest.fn(),
+      },
+      {
+        id: 'b',
+        name: 'b',
+        types: [],
+        dependsOn: ['c'],
+        executionHandler: jest.fn(),
+      },
+    ];
+
+    expect(() => buildStepDependencyGraph(steps)).toThrow(
+      /Node does not exist/,
+    );
+  });
+
+  test('should successfully construct a dependency graph given steps with unique ids and valid dependencies', () => {
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        dependsOn: ['b'],
+        executionHandler: jest.fn(),
+      },
+      {
+        id: 'b',
+        name: 'b',
+        types: [],
+        executionHandler: jest.fn(),
+      },
+    ];
+
+    const graph = buildStepDependencyGraph(steps);
+
+    expect(graph.getNodeData('a')).toEqual(steps[0]);
+    expect(graph.getNodeData('b')).toEqual(steps[1]);
+    expect(graph.dependenciesOf('a')).toEqual(['b']);
+  });
+});
+
+describe('executeStepDependencyGraph', () => {
+  test('executionHandler should have access to "jobState" object', async () => {
+    const executionHandlerSpy = jest.fn();
+
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        executionHandler: executionHandlerSpy,
+      },
+    ];
+
+    const graph = buildStepDependencyGraph(steps);
+    await executeStepDependencyGraph(graph);
+
+    expect(executionHandlerSpy).toHaveBeenCalledTimes(1);
+    const args = executionHandlerSpy.mock.calls[0][0];
+    expect(args.jobState).toBeInstanceOf(FileSystemJobState);
+  });
+
+  test('should perform a flush of the jobState after a step was executed', async () => {
+    let jobStateFlushSpy;
+
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        executionHandler: ({ jobState }) => {
+          jobStateFlushSpy = jest.spyOn(jobState, 'flush');
+        },
+      },
+    ];
+
+    const graph = buildStepDependencyGraph(steps);
+    await executeStepDependencyGraph(graph);
+
+    expect(jobStateFlushSpy).toHaveBeenCalledTimes(1);
+  });
+
+  test('should halt execution of dependant steps until current step has completed', async () => {
```

Comment:
```suggestion
  test('should hold execution of dependent steps until current step has completed', async () => {
```

Reply:
British English vs American English always gets me.

Reply:
Haha! I didn't even consider that when I read it.